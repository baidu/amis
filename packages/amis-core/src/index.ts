export * from './abc';

export function hello() {
  return 'world';
}
