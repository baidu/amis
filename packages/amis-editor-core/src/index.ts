/**
 * @file 总入口
 */
import 'amis';
import './locale/index';
import Editor from './component/Editor';
import './component/ClassNameControl';

import './plugin/AvailableRenderers';
import './plugin/BasicToolbar';
import './plugin/Code';
import './plugin/DataDebug';
import './plugin/ErrorRenderer';
// import './plugin/Name';
import './plugin/Outline';
import './plugin/Unknown';

import * as utils from './util';
export * from './util';
export * from './tpl';
export * from './manager';
export * from './plugin';
export * from './icons/index';
export * from './mocker';
import {BasicEditor, RendererEditor} from './compat';
import MiniEditor from './component/MiniEditor';
import CodeEditor from './component/Panel/AMisCodeEditor';
import IFramePreview from './component/IFramePreview';
import SearchPanel from './component/base/SearchPanel';
import {VRenderer} from './component/VRenderer';
import {RegionWrapper} from './component/RegionWrapper';
import {mapReactElement} from './component/factory';
import type {EditorNodeType, EditorNodeSnapshot} from './store/node';
import {ContainerWrapper} from './component/ContainerWrapper';
import type {EditorStoreType} from './store/editor';
import {AvailableRenderersPlugin} from './plugin/AvailableRenderers';
import ShortcutKey from './component/base/ShortcutKey';
import WidthDraggableContainer from './component/base/WidthDraggableContainer';

export const version = '__buildVersion';

export default Editor;

export {
  Editor,
  MiniEditor,
  utils,
  mapReactElement,
  RendererEditor,
  BasicEditor,
  CodeEditor,
  VRenderer,
  RegionWrapper,
  IFramePreview as IFrameEditor,
  SearchPanel,
  EditorNodeType,
  EditorNodeSnapshot,
  EditorStoreType,
  ContainerWrapper,
  AvailableRenderersPlugin,
  ShortcutKey,
  WidthDraggableContainer
};
