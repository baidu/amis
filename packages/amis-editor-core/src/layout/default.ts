import {LayoutInterface} from './interface';

export default class DefaultLayout implements LayoutInterface {}
