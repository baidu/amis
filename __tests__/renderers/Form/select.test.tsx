import React = require('react');
import {render, screen, fireEvent} from '@testing-library/react';
import '../../../src/themes/default';
import {render as amisRender} from '../../../src/index';
import {makeEnv} from '../../helper';

test('Renderer:select menutpl', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          body: [
            {
              label: '选项',
              type: 'select',
              name: 'select',
              menuTpl:
                '<div>${label} 值：${value}, 当前是否选中: ${checked}</div>',
              options: [
                {
                  label: 'A',
                  value: 'a'
                },
                {
                  label: 'B',
                  value: 'b'
                },
                {
                  label: 'C',
                  value: 'c'
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select group', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '分组',
              type: 'select',
              name: 'a',
              selectMode: 'group',
              options: [
                {
                  label: '法师',
                  children: [
                    {
                      label: '诸葛亮',
                      value: 'zhugeliang'
                    }
                  ]
                },
                {
                  label: '战士',
                  children: [
                    {
                      label: '曹操',
                      value: 'caocao'
                    },
                    {
                      label: '钟无艳',
                      value: 'zhongwuyan'
                    }
                  ]
                },
                {
                  label: '打野',
                  children: [
                    {
                      label: '李白',
                      value: 'libai'
                    },
                    {
                      label: '韩信',
                      value: 'hanxin'
                    },
                    {
                      label: '云中君',
                      value: 'yunzhongjun'
                    }
                  ]
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select table', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '表格形式',
              type: 'select',
              name: 'a',
              selectMode: 'table',
              columns: [
                {
                  name: 'label',
                  label: '英雄'
                },
                {
                  name: 'position',
                  label: '位置'
                }
              ],
              options: [
                {
                  label: '诸葛亮',
                  value: 'zhugeliang',
                  position: '中单'
                },
                {
                  label: '曹操',
                  value: 'caocao',
                  position: '上单'
                },
                {
                  label: '钟无艳',
                  value: 'zhongwuyan',
                  position: '上单'
                },
                {
                  label: '李白',
                  value: 'libai',
                  position: '打野'
                },
                {
                  label: '韩信',
                  value: 'hanxin',
                  position: '打野'
                },
                {
                  label: '云中君',
                  value: 'yunzhongjun',
                  position: '打野'
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select tree', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '树型展示',
              type: 'select',
              name: 'a',
              selectMode: 'tree',
              options: [
                {
                  label: '法师',
                  children: [
                    {
                      label: '诸葛亮',
                      value: 'zhugeliang'
                    }
                  ]
                },
                {
                  label: '战士',
                  children: [
                    {
                      label: '曹操',
                      value: 'caocao'
                    },
                    {
                      label: '钟无艳',
                      value: 'zhongwuyan'
                    }
                  ]
                },
                {
                  label: '打野',
                  children: [
                    {
                      label: '李白',
                      value: 'libai'
                    },
                    {
                      label: '韩信',
                      value: 'hanxin'
                    },
                    {
                      label: '云中君',
                      value: 'yunzhongjun'
                    }
                  ]
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select chained', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '级联选择',
              type: 'select',
              name: 'a',
              selectMode: 'chained',
              options: [
                {
                  label: '法师',
                  children: [
                    {
                      label: '诸葛亮',
                      value: 'zhugeliang'
                    }
                  ]
                },
                {
                  label: '战士',
                  children: [
                    {
                      label: '曹操',
                      value: 'caocao'
                    },
                    {
                      label: '钟无艳',
                      value: 'zhongwuyan'
                    }
                  ]
                },
                {
                  label: '打野',
                  children: [
                    {
                      label: '李白',
                      value: 'libai'
                    },
                    {
                      label: '韩信',
                      value: 'hanxin'
                    },
                    {
                      label: '云中君',
                      value: 'yunzhongjun'
                    }
                  ]
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer: chained select', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '带搜索',
              type: 'select',
              name: 'a',
              selectMode: 'chained',
              searchable: true,
              sortable: true,
              multiple: true,
              options: [
                {
                  label: '法师',
                  children: [
                    {
                      label: '诸葛亮',
                      value: 'zhugeliang'
                    }
                  ]
                },
                {
                  label: '战士',
                  children: [
                    {
                      label: '曹操',
                      value: 'caocao'
                    },
                    {
                      label: '钟无艳',
                      value: 'zhongwuyan'
                    }
                  ]
                },
                {
                  label: '打野',
                  children: [
                    {
                      label: '李白',
                      value: 'libai'
                    },
                    {
                      label: '韩信',
                      value: 'hanxin'
                    },
                    {
                      label: '云中君',
                      value: 'yunzhongjun'
                    }
                  ]
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select associated', () => {
  const {container} = render(
    amisRender(
      {
        type: 'page',
        body: {
          type: 'form',
          api: '/api/mock2/form/saveForm',
          body: [
            {
              label: '关联选择模式',
              type: 'select',
              name: 'b',
              sortable: true,
              searchable: true,
              deferApi: '/api/mock2/form/deferOptions?label=${label}',
              selectMode: 'associated',
              leftMode: 'tree',
              leftOptions: [
                {
                  label: '法师',
                  children: [
                    {
                      label: '诸葛亮',
                      value: 'zhugeliang'
                    }
                  ]
                },
                {
                  label: '战士',
                  children: [
                    {
                      label: '曹操',
                      value: 'caocao'
                    },
                    {
                      label: '钟无艳',
                      value: 'zhongwuyan'
                    }
                  ]
                },
                {
                  label: '打野',
                  children: [
                    {
                      label: '李白',
                      value: 'libai'
                    },
                    {
                      label: '韩信',
                      value: 'hanxin'
                    },
                    {
                      label: '云中君',
                      value: 'yunzhongjun'
                    }
                  ]
                }
              ],
              options: [
                {
                  ref: 'zhugeliang',
                  children: [
                    {
                      label: 'A',
                      value: 'a'
                    }
                  ]
                },
                {
                  ref: 'caocao',
                  children: [
                    {
                      label: 'B',
                      value: 'b'
                    },
                    {
                      label: 'C',
                      value: 'c'
                    }
                  ]
                },
                {
                  ref: 'zhongwuyan',
                  children: [
                    {
                      label: 'D',
                      value: 'd'
                    },
                    {
                      label: 'E',
                      value: 'e'
                    }
                  ]
                },
                {
                  ref: 'libai',
                  defer: true
                },
                {
                  ref: 'hanxin',
                  defer: true
                },
                {
                  ref: 'yunzhongjun',
                  defer: true
                }
              ]
            }
          ]
        }
      },
      {},
      makeEnv({})
    )
  );

  expect(container).toMatchSnapshot();
});

test('Renderer:select virtual', async () => {
  const options = [...Array(200)].map((_, i) => {
    return 'option' + i;
  });

  const {container, queryByText, findByText} = render(
    amisRender(
      {
        type: 'form',
        body: [
          {
            label: '选项',
            type: 'select',
            name: 'select',
            options: options
          }
        ]
      },
      {},
      makeEnv({})
    )
  );
  const inputDate = await findByText('请选择');

  fireEvent.click(inputDate);

  const option12 = queryByText('option12');
  expect(option12).toBeNull();

  expect(container).toMatchSnapshot();
});
