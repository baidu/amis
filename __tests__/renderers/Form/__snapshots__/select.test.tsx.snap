// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer: chained select 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          带搜索
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select associated 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          关联选择模式
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select chained 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          级联选择
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select group 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          分组
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select menutpl 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          选项
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="downshift-0-label"
                      class="cxd-Select"
                      role="combobox"
                      tabindex="0"
                    >
                      <div
                        class="cxd-Select-valueWrap"
                      >
                        <div
                          class="cxd-Select-placeholder"
                        >
                          请选择
                        </div>
                      </div>
                      <span
                        class="cxd-Select-arrow"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select table 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          表格形式
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renderer:select tree 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
        >
          <div
            class="cxd-Panel cxd-Panel--default cxd-Panel--form"
            style="position: relative;"
          >
            <div
              class="cxd-Panel-heading"
            >
              <h3
                class="cxd-Panel-title"
              >
                <span
                  class="cxd-TplField"
                >
                  <span>
                    表单
                  </span>
                </span>
              </h3>
            </div>
            <div
              class="cxd-Panel-body"
            >
              <form
                class="cxd-Form cxd-Form--normal"
                novalidate=""
              >
                <input
                  style="display: none;"
                  type="submit"
                />
                <div
                  class="cxd-Form-item cxd-Form-item--normal"
                  data-role="form-item"
                >
                  <label
                    class="cxd-Form-label"
                  >
                    <span>
                      <span
                        class="cxd-TplField"
                      >
                        <span>
                          树型展示
                        </span>
                      </span>
                    </span>
                  </label>
                  <div
                    class="cxd-SelectControl cxd-Form-control"
                  >
                    <div
                      class="cxd-ResultBox cxd-TransferDropDown cxd-Form-control is-clickable"
                      tabindex="-1"
                    >
                      <span
                        class="cxd-ResultBox-placeholder"
                      >
                        请选择
                      </span>
                      <span
                        class="cxd-TransferDropDown-icon"
                      >
                        <icon-mock
                          classname="icon icon-caret"
                          icon="caret"
                        />
                      </span>
                    </div>
                  </div>
                </div>
              </form>
            </div>
            <div
              class="cxd-Panel-footerWrap"
            >
              <div
                class="cxd-Panel-btnToolbar cxd-Panel-footer"
              >
                <button
                  class="cxd-Button cxd-Button--primary"
                  type="submit"
                >
                  <span>
                    提交
                  </span>
                </button>
              </div>
            </div>
            <div
              class="resize-sensor"
              style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: scroll; z-index: -1; visibility: hidden;"
            >
              
  
              <div
                class="resize-sensor-expand"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 10px; height: 10px;"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-shrink"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;"
              >
                
    
                <div
                  style="position: absolute; left: 0; top: 0; width: 200%; height: 200%"
                />
                
  
              </div>
              
  
              <div
                class="resize-sensor-appear"
                style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: scroll; z-index: -1; visibility: hidden;animation-name: apearSensor; animation-duration: 0.2s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
